lib/i18n-transform.js

```diff
@@ -47,22 +47,13 @@ var _getLanguageFromI18n = function (i18nObj) {
         return _isLanguageCodeMatch(i18nLanguage, language) && _isLanguageRegionMatch(i18nLanguage, language);
     },
 
-    _isPrimaryLanguageMatch = function (i18n, language, primaryLanguageCode) {
-        var i18nLanguage = _getLanguageFromI18n(i18n);
-
-        if (primaryLanguageCode) {
-            if (_isWildcard(language)) {
-                return primaryLanguageCode.toUpperCase() === i18nLanguage.ietf.toUpperCase();
-            } else {
-                return !language.region && primaryLanguageCode.split('-')[0] === language.code;
-            }
-        }
-
-        return _isWildcard(language);
+    _isPrimaryLanguageMatch = function (language, primaryLanguageCode) {
+        return _isWildcard(language) ||
+            !language.region && primaryLanguageCode.split('-')[0] === language.code;
```

Comment:
what if `language` is not wildcard and has `.region`? should you compare it with both parts of `primaryLanguageCode`?